# HG changeset patch
# User Rik <rik@octave.org>
# Date 1430541549 25200
#      Fri May 01 21:39:09 2015 -0700
# Branch stable
# Node ID 3b3579ad7e46b269e99eff7ca9c9aa921b995e80
# Parent  8187a66039e4792a62803af8181f190e6baafae2
Check for a modern version of sndfile in configure scripts (bug #44976).

* configure.ac: Call OCTAVE_CHECK_LIB_SNDFILE_OK after first determining that
sndfile library exists.

* m4/acinclude.m4 (OCTAVE_CHECK_LIB_SNDFILE_OK): New macro to check for a modern
version of lib sndfile with Ogg support.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1015,27 +1015,28 @@ AC_ARG_WITH([sndfile],
    fi],
   [check_sndfile=yes])
 
 if test $check_sndfile = yes; then
   PKG_CHECK_EXISTS([sndfile], [
     SNDFILE_CPPFLAGS=`$PKG_CONFIG --cflags-only-I sndfile`
     SNDFILE_LDFLAGS=`$PKG_CONFIG --libs-only-L sndfile`
     SNDFILE_LIBS=`$PKG_CONFIG --libs-only-l sndfile`
-    warn_sndfile=
+    OCTAVE_CHECK_LIB_SNDFILE_OK([warn_sndfile=])
   ])
 fi
 
 if test -z "$warn_sndfile"; then
   AC_DEFINE(HAVE_SNDFILE, 1, [Define to 1 if sndfile is available.])
 else
   SNDFILE_CPPFLAGS=
   SNDFILE_LDFLAGS=
   SNDFILE_LIBS=
 fi
+
 AC_SUBST(SNDFILE_CPPFLAGS)
 AC_SUBST(SNDFILE_LDFLAGS)
 AC_SUBST(SNDFILE_LIBS)
 
 ### Check for PortAudio
 
 warn_portaudio="PortAudio library fails tests.  The audioplayer, audiorecorder classes and audiodevinfo function for audio playback and recording will not be fully functional."
 
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -1054,16 +1054,38 @@ AC_DEFUN([OCTAVE_CHECK_LIB_QHULL_OK], [
   if test $octave_cv_lib_qhull_ok = yes; then
     $1
     :
   else
     $2
     :
   fi
 ])
+dnl 
+dnl Check whether sndfile library is modern enough to include things like Ogg
+dnl
+AC_DEFUN([OCTAVE_CHECK_LIB_SNDFILE_OK], [
+  AC_CACHE_CHECK([whether sndfile library is modern enough],
+    [octave_cv_lib_sndfile_ok],
+    [AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+        #include <sndfile.h>
+        ]], [[
+        int x = SF_FORMAT_OGG;
+      ]])],
+      octave_cv_lib_sndfile_ok=yes,
+      octave_cv_lib_sndfile_ok=no)
+  ])
+  if test $octave_cv_lib_sndfile_ok = yes; then
+    $1
+    :
+  else
+    $2
+    :
+  fi
+])
 dnl
 dnl Find a suitable termlib to use.
 dnl
 AC_DEFUN([OCTAVE_CHECK_LIB_TERMLIB], [
   TERM_LIBS=
   ac_octave_save_LIBS="$LIBS"
   AC_SEARCH_LIBS([tputs],
                  [ncurses curses termcap terminfo termlib],
